### ISSUE TITLE
SNMP Credentials Issue

### ISSUE THREAD

**Issue by DYKenoby** [glpi-project/glpi-inventory-plugin#112]
After upgrade GLPI 9.5 to GLPI 10.
Unable to normally assosiate snmp credential with ip range.

After click assosiate button empty record appeared and Error in **_log/php-errors.log**
```
`[2022-05-11 19:14:04] glpiphplog.WARNING:   *** PHP Warning (2): Undefined array key "snmpcredentials_id" in /var/www/glpi/marketplace/glpiinventory/inc/iprange_snmpcredential.class.php at line 164
  Backtrace :
  ...entory/inc/iprange_snmpcredential.class.php:114 PluginGlpiinventoryIPRange_SNMPCredential->showItemForm()
  src/CommonGLPI.php:687                             PluginGlpiinventoryIPRange_SNMPCredential::displayTabContentForItem()
  ajax/common.tabs.php:107                           CommonGLPI::displayStandardTab()

[2022-05-11 19:14:04] glpiphplog.WARNING:   *** PHP Warning (2): Undefined array key "snmpcredentials_id" in /var/www/glpi/marketplace/glpiinventory/inc/iprange_snmpcredential.class.php at line 164
  Backtrace :
  ...entory/inc/iprange_snmpcredential.class.php:114 PluginGlpiinventoryIPRange_SNMPCredential->showItemForm()
  src/CommonGLPI.php:687                             PluginGlpiinventoryIPRange_SNMPCredential::displayTabContentForItem()
  ajax/common.tabs.php:107                           CommonGLPI::displayStandardTab()

[2022-05-11 19:14:04] glpiphplog.WARNING:   *** PHP Warning (2): Undefined array key "snmpcredentials_id" in /var/www/glpi/marketplace/glpiinventory/inc/iprange_snmpcredential.class.php at line 221
  Backtrace :
  ...entory/inc/iprange_snmpcredential.class.php:114 PluginGlpiinventoryIPRange_SNMPCredential->showItemForm()
  src/CommonGLPI.php:687                             PluginGlpiinventoryIPRange_SNMPCredential::displayTabContentForItem()
  ajax/common.tabs.php:107                           CommonGLPI::displayStandardTab()

[2022-05-11 19:14:04] glpiphplog.WARNING:   *** PHP Warning (2): Undefined array key "snmpversion" in /var/www/glpi/src/SNMPCredential.php at line 100
  Backtrace :
  ...entory/inc/iprange_snmpcredential.class.php:225 SNMPCredential->getRealVersion()
  ...entory/inc/iprange_snmpcredential.class.php:114 PluginGlpiinventoryIPRange_SNMPCredential->showItemForm()
  src/CommonGLPI.php:687                             PluginGlpiinventoryIPRange_SNMPCredential::displayTabContentForItem()
  ajax/common.tabs.php:107                           CommonGLPI::displayStandardTab()

[2022-05-11 19:14:04] glpiphplog.WARNING:   *** PHP Warning (2): Undefined array key "snmpcredentials_id" in /var/www/glpi/marketplace/glpiinventory/inc/iprange_snmpcredential.class.php at line 221
  Backtrace :
  ...entory/inc/iprange_snmpcredential.class.php:114 PluginGlpiinventoryIPRange_SNMPCredential->showItemForm()
  src/CommonGLPI.php:687                             PluginGlpiinventoryIPRange_SNMPCredential::displayTabContentForItem()
  ajax/common.tabs.php:107                           CommonGLPI::displayStandardTab()

[2022-05-11 19:14:04] glpiphplog.WARNING:   *** PHP Warning (2): Undefined array key "snmpversion" in /var/www/glpi/src/SNMPCredential.php at line 100
  Backtrace :
  ...entory/inc/iprange_snmpcredential.class.php:225 SNMPCredential->getRealVersion()
  ...entory/inc/iprange_snmpcredential.class.php:114 PluginGlpiinventoryIPRange_SNMPCredential->showItemForm()
  src/CommonGLPI.php:687                             PluginGlpiinventoryIPRange_SNMPCredential::displayTabContentForItem()
  ajax/common.tabs.php:107                           CommonGLPI::displayStandardTab()`
```
![Capture1](https://user-images.githubusercontent.com/93933593/167928726-f5e74261-1082-45a2-82e5-28e1d67e2359.PNG)
![Capture2](https://user-images.githubusercontent.com/93933593/167928834-b3903d7b-50dc-4067-8339-2996e622db93.PNG)

**Comment by steph461** [comment]
add user SNMP for exmample  or create user snmp
![image](https://user-images.githubusercontent.com/68054991/168075320-cf374681-02b5-474a-9cad-8c4209b0383b.png)
![image](https://user-images.githubusercontent.com/68054991/168075396-5e2d1039-0533-4fe5-b4c0-d2f0893e1be4.png)

bug migration maybe

**Comment by DYKenoby** [comment]
Good day, and thanks for your answer.
Unfortunatelly result the same:
![image](https://user-images.githubusercontent.com/93933593/168077101-c9910c33-a91c-4f66-a66a-5088489ee01e.png)

**Comment by trasher** [comment]
You probably had an issue during plugin migration. Restore your database, play the migration again; and check logs at this time.

**Comment by DYKenoby** [comment]
Good day.
I found issue, it's look like sql table was not updated during migration...
I performed migration once again on backup image of 9.5 version with the same result.
Then i made clean installation of glpi 10 on new VM, then i found difference in sql database:
Colum in table "glpi_plugin_glpiinventory_ipranges_snmpcredentials" had old name "plugin_glpiinventory_configsecurities_id" in clean version it is "snmpcredentials_id" so i renamed it:  

ALTER TABLE `glpi_plugin_glpiinventory_ipranges_snmpcredentials` CHANGE `plugin_glpiinventory_configsecurities_id` `snmpcredentials_id` INT(10) UNSIGNED NOT NULL DEFAULT '0';

Then new credentials added sucessfully.
![image](https://user-images.githubusercontent.com/93933593/169062523-a05a9334-1cc6-4dee-a6c8-034cc2aaaeb0.png)
